applications/FluidDynamicsApplication/custom_elements/alternative_d_vms_dem_coupled.cpp

```diff
@@ -334,62 +698,66 @@ void AlternativeDVMSDEMCoupled<TElementData>::MomentumProjTerm(
+    for (unsigned int i = 0; i < NumNodes; i++)
+        for (unsigned int d = 0; d < Dim; d++)
+            fluid_fraction_gradient[d] += rData.DN_DX(i,d) * rData.FluidFraction[i];
+    const auto& body_force = this->GetAtCoordinate(rData.BodyForce, rData.N);
+    Vector grad_alpha_sym_grad_u, grad_div_u, sigma_U, div_sym_grad_u;
+    BoundedMatrix<double,Dim,Dim> sym_gradient_u;
     for (unsigned int i = 0; i < NumNodes; i++) {
-        sigma_u = ZeroVector(Dim);
-        BoundedMatrix<double,Dim,Dim> sym_gradient_u = ZeroMatrix(Dim, Dim);
+
+        grad_div_u = ZeroVector(Dim);
+        sym_gradient_u = ZeroMatrix(Dim, Dim);
         grad_alpha_sym_grad_u = ZeroVector(Dim);
+        div_sym_grad_u = ZeroVector(Dim);
         for (unsigned int d = 0; d < Dim; d++) {
             double div_u = 0.0;
             for (unsigned int e = 0; e < Dim; e++){
-                sigma_u[d] += sigma(d,e) * rData.N[i] * rData.Velocity(i,e);
                 sym_gradient_u(d,e) += 1.0 / 2.0 * (rData.DN_DX(i,d) * rData.Velocity(i,e) + rData.DN_DX(i,e) * rData.Velocity(i,d));
-                grad_alpha_sym_grad_u[d] += fluid_fraction_gradient[d] * sym_gradient_u(d,e);
+                grad_alpha_sym_grad_u[d] += fluid_fraction_gradient[e] * sym_gradient_u(d,e);
                 div_u += rData.DN_DX(i,e) * rData.Velocity(i,e);
+                grad_div_u[d] += rData.DDN_DDX[i](d,e) *  rData.Velocity(i,e);
+                if (d == e)
+                    div_sym_grad_u[d] += rData.DDN_DDX[i](e,e) * rData.Velocity(i,d);
+                else
+                    div_sym_grad_u[d] += 1.0/2.0 * (rData.DDN_DDX[i](e,d) * rData.Velocity(i,e) + rData.DDN_DDX[i](e,e) * rData.Velocity(i,d));
             }
-            rMomentumRHS[d] += density * ( rData.N[i] * (rData.BodyForce(i,d)) /*- fluid_fraction *  rData.N[i] * rAcc[d]*/
-                                - fluid_fraction * AGradN[i] * rData.Velocity(i,d)) + 2 * grad_alpha_sym_grad_u[d] * viscosity - 2.0 / 3.0 * viscosity * fluid_fraction_gradient[d] * div_u - fluid_fraction * rData.DN_DX(i,d)*rData.Pressure[i] - sigma_u[d];
+            rMomentumRHS[d] += density * (- fluid_fraction * AGradN[i] * rData.Velocity(i,d)) + 2.0 * grad_alpha_sym_grad_u[d] * viscosity - 2.0/3.0 * viscosity * fluid_fraction_gradient[d] * div_u + 2.0 * fluid_fraction * viscosity * div_sym_grad_u[d] - 2.0/3.0 * fluid_fraction * viscosity * grad_div_u[d] - fluid_fraction * rData.DN_DX(i,d) * rData.Pressure[i]/* - sigma_U[d]*/;
         }
     }
+    for (unsigned int d = 0; d < Dim; d++)
+        rMomentumRHS[d] += density * body_force[d];
 }
 
+// Add a the contribution from a single integration point to the velocity contribution
 template< class TElementData >
 void AlternativeDVMSDEMCoupled<TElementData>::AddVelocitySystem(
     TElementData& rData,
     MatrixType &rLocalLHS,
     VectorType &rLocalRHS)
 {
-    auto& LHS = rData.LHS;
+    BoundedMatrix<double,NumNodes*(Dim+1),NumNodes*(Dim+1)>& LHS = rData.LHS;
```

Comment:
why not using the previous `auto`? it was clearer

Reply:
It produces compilation errors, since this variable has to be called in `AddReactionStabilizationTerms` function. You cannot pass it as `auto`